modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-instance/src/main/java/com/liferay/dynamic/data/mapping/data/provider/instance/DDMStorageTypesDataProvider.java

```diff
@@ -43,11 +48,31 @@ public DDMDataProviderResponse getData(
 
 		List<KeyValuePair> keyValuePairs = new ArrayList<>();
 
+		Optional<HttpServletRequest> httpServletRequestOptional =
+			ddmDataProviderRequest.getParameterOptional(
+				"httpServletRequest", HttpServletRequest.class);
+
+		HttpServletRequest httpServletRequest =
+			httpServletRequestOptional.orElse(null);
+
 		Set<String> storageTypes =
 			ddmStorageAdapterTracker.getDDMStorageAdapterTypes();
 
 		for (String storageType : storageTypes) {
-			keyValuePairs.add(new KeyValuePair(storageType, storageType));
+			if (httpServletRequest == null) {
+				keyValuePairs.add(new KeyValuePair(storageType, storageType));
+
+				continue;
+			}
+
+			String value = LanguageUtil.get(httpServletRequest, storageType);
+
+			if (storageType.equals(StorageType.JSON.toString())) {
+				value = LanguageUtil.get(
+					httpServletRequest, "json[stands-for]", storageType);
+			}
```

Comment:
@carolmariaabb, can we create a more generic rule to handle all language keys that may exist with `[stand-for]`? Otherwise, we'll need to create an `if` for each new storage type which uses `[stand-for]`.

**Tip:** when there is no value associated to a specific language key, then the language key is returned from the method `LanguageUtil.get`. 😉